go/mysql/bufio_pool.go

```diff
@@ -0,0 +1,113 @@
+package mysql
+
+import (
+	"bufio"
+	"io"
+	"sync"
+)
+
+// Reader implementation
+
+type bufioReader interface {
+	Read([]byte) (int, error)
+	Reset(io.Reader)
+}
+
+var readersPool = sync.Pool{New: func() interface{} { return bufio.NewReaderSize(nil, connBufferSize) }}
+
+func (pbr *poolBufioReader) getReader() {
+	if pbr.br != nil {
+		return
+	}
+	pbr.br = readersPool.Get().(*bufio.Reader)
+	pbr.br.Reset(pbr.r)
+}
+
+func (pbr *poolBufioReader) putReader() {
+	if pbr.br == nil {
+		return
+	}
+	// remove reference
+	pbr.br.Reset(nil)
+	readersPool.Put(pbr.br)
+	pbr.br = nil
+}
+
+type poolBufioReader struct {
+	r  io.Reader
+	br *bufio.Reader
+}
+
+func newReader(r io.Reader) bufioReader {
+	return &poolBufioReader{
+		r: r,
+	}
+}
+
+func (pbr *poolBufioReader) Read(b []byte) (int, error) {
+	pbr.getReader()
```

Comment:
so for the time being, this is going to eventually cause all connections to accumulate a reader, since we never Reset() them